src/braket/default_simulator/openqasm/_helpers/functions.py

```diff
@@ -98,10 +98,16 @@
             [BooleanLiteral(xv.value ^ yv.value) for xv, yv in zip(x.values, y.values)]
         ),
         getattr(BinaryOperator, "<<"): lambda x, y: ArrayLiteral(
-            x.values[y.value :] + [BooleanLiteral(False) for _ in range(y.value)]
+            x.values[len(y.values) :] + [BooleanLiteral(False) for _ in range(len(y.values))]
+            if isinstance(y, ArrayLiteral)
+            else x.values[y.value :] + [BooleanLiteral(False) for _ in range(y.value)]
```

Comment:
[minor] this inline logic is getting hard to read - could it be factored out into an internal helper function? perhaps `casting.py` would be a good place.